### PR TITLE
Feature/dl 42

### PR DESCRIPTION
**Pull request by lukasz-jaszewski**
Decorators for handling users roles, there are 3 decorators:
* **require_logged_in(redirect_page="library.login")** - checks if user logged in, if not redirects
* **require_not_logged_in(redirect_page="library.index")** - checks if user logged in, if yes redirects
* **require_role(role="USER", redirect_page="library.index")** - checks user role, if it is not the same as 'role' variable then redirects. There's also accepted role = 'ADMIN'  or any other string.

Changed files:
* models/decorators_roles.py - decorators handling users roles
* tests/test_decorators_roles.py - test for decorators
* **send_email/confirm.py** - fixed confirmation mail (don't apply to this task)
* **forms/registration_forms.py** - deleted, duplicate of other form (don't apply to this task)

### CODE REVIEW

File: models/decorators_roles.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    return func(*args, **kwargs)
+                else:
+                    return redirect(url_for(redirect_page))
+
+            else:
+                return redirect(url_for(redirect_page))
+
+        return inner_func
+
+    return decorator
```
Comment (kuszcjan) [inline models/decorators_roles.py]: Please remove unnecessary blank lines.

File: requirements.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@ flask-login
 flask_mail==0.9.1
 Flask-Session
 Flask-SQLAlchemy==2.3.1
-flask-user
+Flask-User<0.7
```
Comment (kuszcjan) [inline requirements.txt]: please elaborate on this change.

File: models/decorators_roles.py (hunk 1 above)
Comment (Jahenstein) [reply models/decorators_roles.py]: Thanks, I'll work on this

File: requirements.txt (hunk 2 above)
Comment (Jahenstein) [reply requirements.txt]: Newest version - Flask-user 1.0, is alpha release. Additionally it is incompatible with older version - Flask-user 0.6.*, and older release have slightly better community support ;D

Comment (oskam) [reply requirements.txt]: Are we using it in app anyway?

**Comment by mssliw** [comment]
I am not sure why you have made these formatting changes in `xlsx_reader.py` . As far as I can see it is not associated with the task you deliver, like many other files you have edited. For example, I am still working on this file and fixing some bugs, so I am afraid we may have unnecessary conflicts during the merge. I am also not sure if these changes were necessary, especially because Stickler must have suggested changes while these files were delivered.

I would be grateful if another reviewer would consider if these changes may affect the merge.

I recommend using Flake 8 for checking files which are in scope of your task.  This tool is very useful, works for me. :)

**Comment by kuszcjan** [comment]
In my opinion its bad to make global changes in the code, unless this is specifically the task. Such approach only introduces more chaos.

Changing globally double quotes instead of single ones provide no value to the project.

First of all, leave double quotes as they were.
Second, we agreed to fix obvious bugs if encounter during implementation. So focus only on area related to PR. Do not change other code without reason.

File: requirements.txt (hunk 2 above)
Comment (Jahenstein) [reply requirements.txt]: Yes, in models/users.py

**Comment by Jahenstein** [comment]
I experimented with Black, it edited all files in project. Sorry for mess, I'll clean this up.

File: models/decorators_roles.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+from flask import session, redirect, url_for
+from functools import wraps
+from models.users import User
+
+
+def require_logged_in(redirect_page='library.login'):
```
Comment (etude-ist) [inline models/decorators_roles.py]: I have Déjà-vu. ;)

**Pull request merged**